include/matx/transforms/matmul.h

```diff
@@ -997,31 +1035,92 @@ static matxCache_t<MatMulParams_t, MatMulParamsKeyHash, MatMulParamsKeyEq>
  */
 template <typename TensorTypeC, typename TensorTypeA, typename TensorTypeB, 
           MatXMatMulProvider_t PROV = PROVIDER_TYPE_CUBLASLT>
-void matmul(TensorTypeC &c, const TensorTypeA &a,
-            const TensorTypeB &b, cudaStream_t stream = 0,
+void matmul(TensorTypeC c, const TensorTypeA a,
+            const TensorTypeB b, cudaStream_t stream = 0,
             float alpha = 1.0, float beta = 0.0)
 {
   MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)
-  
-  // Get parameters required by these tensors
-  auto params =
+ 
+#if MATX_ENABLE_CUTLASS != 1
+  // cublas does not support column major output.  Use the matrix multiply of transposes property to make output row major.
```

Comment:
This comment probably needs more context? cuBLAS is only column major. Did you mean cuBLASLt in row major mode doesn't support col-major output?

Reply:
fixed